Update the WAN Assurance release and re-onboarding workflow (#415)
128technology/docs: docs/config_wan_assurance.md

Once the site assignment is complete, the information is relayed back to the corresponding SSR router, and the router begins streaming the telemetry data to the cloud.

### Viewing the SSR status

The status of the SSR connection to the MIST cloud and other details can be found on the GUI and PCLI.

#### On the Conductor UI

* Navigate to Plugins > MIST WAN ASSURANCE > Details.

 ![Plugins Context Menu](/img/wana_plugin_detail.png)

* Select the router from the `Router Context` menu.

 ![Plugins Context Menu](/img/wana_plugin_context.png)

* The summary output is displayed by default and contains useful information about the device connection to the MIST cloud.

 ![Plugins Context Menu](/img/wana_plugin_state.png)

#### On the Conductor PCLI

* The `show mist` command can be used to display the summary and detail information about the 128 router's connection to the MIST cloud.

```console
admin@node1.conductor1# show mist router SSR-east
Wed 2022-07-20 05:33:05 UTC
✔ Retrieving mist state...
Target: node1.SSR-east

========== ============ ============= ================= ============================= =================== ============

>  Agent      Connection   SSR          SSR-mist-agent   Platform                      Device-ID           Interfaces

Reviewer: 128T or SSR?

Author: the agent is called 128T-mist-agent so this is just reflecting that. I'll open a JIRA to update it but since this is the CLI output copy/paste I think it should say this for now.

Author: oops - I see the problem! - sorry - i misread!

Once the site assignment is complete, the information is relayed back to the corresponding SSR router, and the router begins streaming the telemetry data to the cloud.

> ### Viewing the current status (128T)

Reviewer 2: ```suggestion
### Viewing the SSR status 
```

### Show Command
The `show plugins state` command displays extensive information about the state of the plugin. Use the following command as a starting point for any troubleshooting.

> `show plugins state router <router name> summary SSR-mist-wan-assurance`

Reviewer: This is a pcli command, it must match what the product has for syntax.

The `connection` column will display the current status of the device connection to the MIST cloud. When the connection is down, it will also display information useful to diagnose the connection issue.

```
admin@node1.office# show plugins state summary SSR-mist-wan-assurance
Thu 2022-05-12 22:38:34 EDT
✔ Retrieving state data...
Target: node1.office

============ ======================== ============== ================= =================== ============

>  Agent        Connection               SSR           SSR-mist-agent   Device-ID           Interfaces

Reviewer: Since this is PCLI output, it should match what the system outputs. Are these headings actually "SSR" or "128T"?

For additional troubleshooting information use the `detail` commands as shown below

> `show plugins state router <router name> detail SSR-mist-wan-assurance`

Reviewer: same as above

### Releasing a Router

> To release a router, use the following process from the MIST GUI:

Author: > First question: is it no longer required to release the router from BOTH Mist and the conductor

Correct - only release from MIST portal. The `release mist agent` command has been removed for a while now but the doc was never updated.

>  Is it still required to release the conductor from both Mist and SSR in the earlier version of the software? 

Conductor is not onboarded to the MIST cloud

This section was incorrect and confusing pretty much from the get go.

* Go to **Organization > Inventory**.
* Select the checkbox next to the router.
* Click on the **More** drop down that appears on the top of the page.
* Select **Release**.
* Click on **Release** to confirm.

After a few minutes the router will be released and stop sending the data to the Mist cloud. Follow the instructions in [this section](config_wan_assurance#viewing-the-current-status-128t) to check the current status. For a successfully released router, the output would look as follows

 ![Plugins Context Menu](/img/wana_plugin_released.png)

Once that process is completed, disable the Mist WAN Assurance plugin on the router.

![Disable MWA Plugin](/img/wana_disabled_mwa.png)

### Re-Onboarding a Router

Once the necessary changes have been made to the registration-code and other configuration, the re-onboarding process can begin.

#### On the 128T conductor UI

* Navigate to Plugins > MIST WAN ASSURANCE > Details

 ![Plugins Context Menu](/img/wana_plugin_detail.png)

> * Confirm the registration-code on the router is as expected. This can done via the `Detail State` tab as shown below

Reviewer 2: ```suggestion
2. Confirm the registration-code on the router is as expected using the `Detail State` tab as shown below.
```

>  ![Plugins Detail State](/img/wana_plugin_detail_regcode.png)

Author: yes!

> * Re-enable the plugin on the router and commit the configuration

Reviewer 2: ```suggestion
3. Re-enable the plugin on the router and commit the configuration.
```

![Enable MWA Plugin](/img/wana_enabled_mwa.png)

> * Switch to the `Command` tab, select the router from the `Router Context` menu and select the `unrelease mist agent` command

Reviewer 2: ```suggestion
4. Switch to the `Command` tab, select the router from the `Router Context` menu, and select the `unrelease mist agent` command.
```

 ![Plugins Command Context Menu](/img/wana_plugin_command_context.png)

> * Click on `Execute` to initiate the re-onboarding process using the registration-code currently available on the system.

Reviewer 2: ```suggestion
5. Click on `Execute` to initiate the re-onboarding process using the registration-code currently available on the system.
```

 ![Unreleased](/img/wana_unreleased.png)

> #### On the conductor PCLI

Reviewer 2: ```suggestion
#### Using the Conductor PCLI
```

* Confirm the registration-code on the router is as expected

```console
admin@node1.conductor1# show mist router 128t-east detail
Wed 2022-07-20 05:58:33 UTC
✔ Retrieving mist state...

==============================================================================================================================================================================================================================================================
 node1.128t-east
==============================================================================================================================================================================================================================================================
   128T:                                       5.6.1-9
   128T-mist-agent:                            4.0.3330-1
   Agent:                                      released
   Connection:                                 down

>    registration-code:                          eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJvcmdfaWQiOiIwYzE2MGI3Zi0xMDI3LTRjZDEtOTIzYi03NDQ1MzRjNGIwNzAiLCJzdmMiOiIxMjhyb3V0ZXIiLCJwcm92aWRlciI6ImF3cyIsImVudiI6InN0YWdpbmciLCJlcHRlcm1fdXJsIjoid3NzOi8vZXAtdGVyb

Author: I've modified it at the end such that it won't be usable.

 WluYXRvci1zdGFnaW5nLm1pc3RzeXMubmV0L3dzIiwiaWF0IjoxNjU4MjkzMTQ2LCJleHAiOjE2ODk4MjkxNDZ9.LVIW0Gx8Q8IIkp0o1J86ZoW3_FytrRDKJEMQdBqABCD
   registration-time:                          unknown

Retrieved mist state.
Completed in 0.97 seconds
admin@node1.conductor1#
```

> * Re-enable the plugon on the router

Reviewer 2: ```suggestion
2. Re-enable the plugin on the router.
```

```console
admin@node1.conductor1# config authority router 128t-east mist-wan-assurance enabled true
```

> * Unrelease the router to initiate the re-onboarding process

Reviewer 2: ```suggestion
3. Unrelease the router to initiate the re-onboarding process.
```

```console
admin@node1.conductor1# unrelease mist agent
node   router
admin@node1.conductor1# unrelease mist agent router 128t-east node node1
✔ Retrieving...
Target: node1.conductor1

Unreleased

Successfully retrieved info.
admin@node1.conductor1#
```

> * Once re-onboarded, the status can be inspected via the `show mist` command

Reviewer 2: ```suggestion
3. Once re-onboarded, check the status using the `show mist` command.
```

![Enable MWA Plugin](/img/wana_enabled_mwa.png)

4. Switch to the `Command` tab, select the router from the `Router Context` menu and select the `unrelease mist agent` command.

 ![Plugins Command Context Menu](/img/wana_plugin_command_context.png)

5. Click on `Execute` to initiate the re-onboarding process using the registration-code currently available on the system.

 ![Unreleased](/img/wana_unreleased.png)

#### Using the conductor PCLI

1. Confirm the registration-code on the router is as expected.

```console
admin@node1.conductor1# show mist router SSR-east detail
Wed 2022-07-20 05:58:33 UTC
✔ Retrieving mist state...

==============================================================================================================================================================================================================================================================
 node1.SSR-east
==============================================================================================================================================================================================================================================================

>    SSR:                                       5.6.1-9

Reviewer: This output needs to match the product output

Author: same - the CLI is what it is right now. Once we fix it we can update the doc

admin@node1.conductor1#
```

4. Once re-onboarded, check the status using the `show mist` command.
```console
admin@node1.conductor1# show mist router SSR-east
Wed 2022-07-20 06:01:32 UTC
✔ Retrieving mist state...
Target: node1.SSR-east

============ ============ ============= ================= ============================= =================== ============

>  Agent        Connection   SSR          SSR-mist-agent   Platform                      Device-ID           Interfaces

Reviewer: Is this output correct that the heading is "SSR"?

Author: same as above
